### ISSUE TITLE
Implement GB support in SMIRFF

### ISSUE THREAD

**Issue by jchodera** [open-forcefield-group/smarty#165]
We can easily add in GBSA support. 

Perhaps syntax like this?

``` XML
<ForceField>
 <GBSAForce model="OBC1" radius_units="nanometers" sasa_penalty="5" sasa_penalty_units="cal/mol/angstroms**2">
   <Atom smarts="[H:1]" radius="0.12" scale="0.85"/>
   <Atom smarts="[H:1]~C" radius="0.13" scale="0.85"/>
   <Atom smarts="[H:1]~O" radius="0.08" scale="0.85"/>
   <Atom smarts="[H:1]~S" radius="0.08" scale="0.85"/>
   <Atom smarts="C" radius="0.22" scale="0.72"/>
   <Atom smarts="N" radius="0.155" scale="0.79"/>
   <Atom smarts="O" radius="0.15" scale="0.85"/>
   <Atom smarts="F" radius="0.15" scale="0.88"/>
   <Atom smarts="Si" radius="0.21" scale="0.8"/>
   <Atom smarts="P" radius="0.185" scale="0.86"/>
   <Atom smarts="S" radius="0.18" scale="0.96"/>
   <Atom smarts="Cl" radius="0.17" scale="0.8"/>
 </GBSAForce>
</ForceField>
```

That's actually a complete specification of the `iGBparm=1` parameters, I believe.

**Comment by davidlmobley** [comment]
Looks mostly good, except you're mixing SMARTS and SMIRKS (i.e. the first few are SMIRKS, not SMARTS; whereas the latter few are SMARTS). For consistency with everything else you should just be using SMIRKS.

Note also that even your SMARTS have issues -- SMARTS is based on SMILES, so `C` gets you only a subset of carbons (non-aromatic carbons, since aromatics are `c`) and likewise for `N`, `O`, and `S`. So here you'd be missing parameters for aromatic carbon, nitrogen, oxygen, and sulfur.

So, we should (a) move the whole thing to SMIRKS, and (b) move to atomic numbers at least for the cases where SMILES/SMARTS distinguish between aromatic and non-aromatic, so you at least want:

``` XML
<ForceField>
 <GBSAForce model="OBC1" radius_units="nanometers" sasa_penalty="5" sasa_penalty_units="cal/mol/angstroms**2">
   <Atom smirks="[H:1]" radius="0.12" scale="0.85"/>
   <Atom smirks="[H:1]~[#6]" radius="0.13" scale="0.85"/>
   <Atom smirks="[H:1]~[#8]" radius="0.08" scale="0.85"/>
   <Atom smirks="[H:1]~[#16]" radius="0.08" scale="0.85"/>
   <Atom smirks="[#6:1]" radius="0.22" scale="0.72"/>
   <Atom smirks="[#7:1]" radius="0.155" scale="0.79"/>
   <Atom smirks="[#8:1]" radius="0.15" scale="0.85"/>
   <Atom smirks="[F:1]" radius="0.15" scale="0.88"/>
   <Atom smirks="[Si:1]" radius="0.21" scale="0.8"/>
   <Atom smirks="[P:1]" radius="0.185" scale="0.86"/>
   <Atom smirks="[#16:1]" radius="0.18" scale="0.96"/>
   <Atom smirks="[Cl:1]" radius="0.17" scale="0.8"/>
 </GBSAForce>
</ForceField>
```

Once we've committed to using atomic numbers in some places (as we must for reasons noted above), I prefer to go the whole way and use atomic numbers everywhere (though Christopher and I differ on this point) so this is my preferred form:

``` XML
<ForceField>
 <GBSAForce model="OBC1" radius_units="nanometers" sasa_penalty="5" sasa_penalty_units="cal/mol/angstroms**2">
   <Atom smirks="[#1:1]" radius="0.12" scale="0.85"/>
   <Atom smirks="[#1:1]~[#6]" radius="0.13" scale="0.85"/>
   <Atom smirks="[#1:1]~[#8]" radius="0.08" scale="0.85"/>
   <Atom smirks="[#1:1]~[#16]" radius="0.08" scale="0.85"/>
   <Atom smirks="[#6:1]" radius="0.22" scale="0.72"/>
   <Atom smirks="[#7:1]" radius="0.155" scale="0.79"/>
   <Atom smirks="[#8:1]" radius="0.15" scale="0.85"/>
   <Atom smirks="[#9:1]" radius="0.15" scale="0.88"/>
   <Atom smirks="[#14:1]" radius="0.21" scale="0.8"/>
   <Atom smirks="[#15:1]" radius="0.185" scale="0.86"/>
   <Atom smirks="[#16:1]" radius="0.18" scale="0.96"/>
   <Atom smirks="[#17:1]" radius="0.17" scale="0.8"/>
 </GBSAForce>
</ForceField>
```

(However, both should parse properly so this is just a style issue.)

**Comment by jchodera** [comment]
Thanks for the fixes!

One more question: Do you prefer

```
sasa_penalty="5" sasa_penalty_units="cal/mol/angstroms**2"
```

or

```
sasa_penalty="5*cal/mol/angstroms**2"
```

I think both could work in our current system, though I think the former form is probably more consistent with our current approach.

**Comment by davidlmobley** [comment]
I agree with you that the former is more consistent -- we tend to specify units and values separately. So I'd go with that.

**Comment by davidlmobley** [comment]
@jchodera - is there any possibility of you doing this in the near future so we can proceed towards separating SMIRNOFF from `smarty`? Otherwise we should perhaps go ahead and do so without GBSA support. Or perhaps I might be able to figure out how to do this with my group.

**Comment by jchodera** [comment]
Yup, sorry, had written a response but forgot to press `Comment`. Let's just finish this now and then we can move on with pulling SMIRNOFF out.

**Comment by jchodera** [comment]
I'll try to finish this up tonight.

**Comment by davidlmobley** [comment]
Any headway, @jchodera ?

**Comment by davidlmobley** [comment]
Resolved by #182 .
